docs/README.vlan.bridging.topo.md

```diff
@@ -0,0 +1,266 @@
+
+Pre-requisites for testing this out
+    Make sure the docker is installed on the Linux system. iproute2 and sudo packages should be installed.
+    
+Load container images
+```
+docker load < docker-sonic-vpp.gz
+```
+
+Check if images are listed 
+```
+docker images | grep "sonic-vpp\|pause"
+```
+
+Get the script to create and run the sonic container from the sonic-platform-vpp repo if you have not cloned the repo.
+```
+wget https://raw.githubusercontent.com/sonic-net/sonic-platform-vpp/main/start_sonic_vpp.sh 
+```
+Set the execute bits of the downloaded script file
+```
+chmod +x start_sonic_vpp.sh
+
+```
+Else if you have already cloned the sonic-platform-repo you can copy the script to the local directory and use it.
+
+On the host create veth interface pairs for vpp and host 
+
+```
+sudo ip link add name veth_ac1 type veth peer name ac1
+sudo ip link add name veth_ac2 type veth peer name ac2
+sudo ip link add name veth_ac3 type veth peer name ac3
+sudo ip link add name veth_ac4 type veth peer name ac4
+sudo ip link add name veth_ac5 type veth peer name ac5
+sudo ip link add name veth_ac6 type veth peer name ac6
+sudo ip link add name veth_trunk type veth peer name trunk
+
+```
+
+Create network namespace for end hosts and inject one end of veth pair
+
+```
+sudo ip netns add host-1.0
+sudo ip netns add host-2.0
+sudo ip netns add host-3.0
+sudo ip netns add host-4.0
+sudo ip netns add host-5.0
+sudo ip netns add host-6.0
+sudo ip link set dev veth_ac1 netns host-1.0
+sudo ip link set dev veth_ac2 netns host-2.0
+sudo ip link set dev veth_ac3 netns host-3.0
+sudo ip link set dev veth_ac4 netns host-4.0
+sudo ip link set dev veth_ac5 netns host-5.0
+sudo ip link set dev veth_ac6 netns host-6.0
+```
+
+Configure IP addresses inside the host net namepaces
+
+```
+sudo ip netns exec host-1.0 bash
+ip link set dev veth_ac1 up
+ip addr add 10.0.1.1/24 dev veth_ac1
+ip address show
+ip route show
+exit
+sudo ip netns exec host-2.0 bash
+ip link set dev veth_ac2 up
+ip addr add 10.0.2.1/24 dev veth_ac2
+ip address show
+ip route show
+exit
+sudo ip netns exec host-3.0 bash
+ip link set dev veth_ac3 up
+ip addr add 10.0.3.1/24 dev veth_ac3
+ip address show
+ip route show
+exit
+sudo ip netns exec host-4.0 bash
+ip link set dev veth_ac4 up
+ip addr add 10.0.1.2/24 dev veth_ac4
+ip address show
+ip route show
+exit
+sudo ip netns exec host-5.0 bash
+ip link set dev veth_ac5 up
+ip addr add 10.0.2.2/24 dev veth_ac5
+ip address show
+ip route show
+exit
+sudo ip netns exec host-6.0 bash
+ip link set dev veth_ac6 up
+ip addr add 10.0.3.2/24 dev veth_ac6
+ip address show
+ip route show
+exit
+```
+
+Now start the sonic container and pass the veth interfaces to sonic-vpp
+```
+sudo ./start_sonic_vpp.sh start -n sonic-br1 -i ac1,ac2,ac3,trunk
+```
+
+start the second sonic container and pass the veth interfaces to sonic-vpp
+```
+sudo ./start_sonic_vpp.sh start -n sonic-br2 -i ac3,ac4,ac5,veth_trunk
```

Comment:
-i argument values should be ac4,ac5,ac6,veth_trunk
right?

Reply:
Accepted